src/main/fc/rc.c

```diff
@@ -344,37 +344,36 @@ bool getRxRateValid(void)
-    const float minCutoffHz = 15.0f; // don't let any RC smoothing filter cutoff go below 15Hz
-
     const bool autoSetpointSmoothing = smoothingData->setpointCutoffSetting == 0;
     const bool autoThrottleSmoothing = smoothingData->throttleCutoffSetting == 0;
 
-    float setpointCutoffFrequency = smoothingData->setpointCutoffFrequency;
-    float throttleCutoffFrequency = smoothingData->throttleCutoffFrequency;
-    if (autoSetpointSmoothing) {
-        setpointCutoffFrequency = MAX(minCutoffHz, smoothedRxRateHz * smoothingData->autoSmoothnessFactorSetpoint);
-        smoothingData->setpointCutoffFrequency = setpointCutoffFrequency; // update for logging
-    }
+    const float minCutoffHz = 15.0f; // don't let any RC smoothing filter cutoff go below 15Hz
+    const float dT = targetPidLooptime * 1e-6f;
 
-    if (autoThrottleSmoothing) {
-        throttleCutoffFrequency = MAX(minCutoffHz, smoothedRxRateHz * smoothingData->autoSmoothnessFactorThrottle);
-        smoothingData->throttleCutoffFrequency = throttleCutoffFrequency; // update for logging
-    }
+    if (autoSetpointSmoothing) {
+        float setpointCutoffFrequency = MAX(minCutoffHz, smoothedRxRateHz * smoothingData->autoSmoothnessFactorSetpoint);
 
-    const float dT = targetPidLooptime * 1e-6f;
+        smoothingData->setpointCutoffFrequency = setpointCutoffFrequency;
 
-    // Update the RC Setpoint/Deflection filter and FeedForward Filter
-    // all cutoffs will be the same, we can optimize :)
-    const float pt3K = pt3FilterGain(setpointCutoffFrequency, dT);
-    for (int i = FD_ROLL; i <= FD_YAW; i++) {
-        pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[i], pt3K);
-        pt3FilterUpdateCutoff(&smoothingData->filterFeedforward[i], pt3K);
-    }
-    for (int i = FD_ROLL; i <= FD_PITCH; i++) {
-        pt3FilterUpdateCutoff(&smoothingData->filterRcDeflection[i], pt3K);
+        const float pt3K_SP = pt3FilterGain(setpointCutoffFrequency, dT);
+        for (int i = FD_ROLL; i <= FD_YAW; i++) {
+            pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[i], pt3K_SP);
+            pt3FilterUpdateCutoff(&smoothingData->filterFeedforward[i], pt3K_SP);
+        }
     }
 
-    pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[THROTTLE], pt3FilterGain(throttleCutoffFrequency, dT));
+    if (autoThrottleSmoothing) {
+        float throttleCutoffFrequency = MAX(minCutoffHz, smoothedRxRateHz * smoothingData->autoSmoothnessFactorThrottle);
+        smoothingData->throttleCutoffFrequency = throttleCutoffFrequency;
+        
+        // Update throttle filter with its own cutoff
+        const float pt3K_Thr = pt3FilterGain(throttleCutoffFrequency, dT);
+        pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[THROTTLE], pt3K_Thr);
+        
+        for (int i = FD_ROLL; i <= FD_PITCH; i++) {
+            pt3FilterUpdateCutoff(&smoothingData->filterRcDeflection[i], pt3K_Thr);
```

Comment:
I'm not sure if this is on purpose but all 4 stick controls (roll, pitch, yaw, throttle) ALL get initialized with the throttle filter cutoff. After reading the comment `// Update throttle filter with its own cutoff` I guess that this is actually not intended.

Reply:
Updated. See also comment below